crank/bin/mcd_calculate.py

```diff
@@ -0,0 +1,166 @@
+        else:
+            cvt_mcep, cvt_f0 = get_world_features(cvt_path, trgspk, conf, spkr_conf)
+
+        # get ground truth features
+        gt_mcep = read_feature(gt_file_list[f"{trgspk}_{number}"], "mcep")
+        gt_f0 = read_feature(gt_file_list[f"{trgspk}_{number}"], "f0")
+
+        # non-silence parts
+        gt_idx = np.where(gt_f0>0)[0]
+        gt_mcep = gt_mcep[gt_idx]
+        cvt_idx = np.where(cvt_f0>0)[0]
+        cvt_mcep = cvt_mcep[cvt_idx]
+
+        # DTW
+        _, path = fastdtw(cvt_mcep, gt_mcep, dist=scipy.spatial.distance.euclidean)
+        twf = np.array(path).T
+        cvt_mcep_dtw = cvt_mcep[twf[0]]
+        gt_mcep_dtw = gt_mcep[twf[1]]
+
+        # MCD
+        diff2sum = np.sum((cvt_mcep_dtw - gt_mcep_dtw)**2, 1)
+        mcd = np.mean(10.0 / np.log(10.0) * np.sqrt(2 * diff2sum), 0)
+        MCD[f"{srcspk}-{trgspk}-{number}"] = mcd
+
+def main():
+    
+    parser = argparse.ArgumentParser(
+        description="calculate MCD.")
+    parser.add_argument('--conf', type=str, required=True,
+                        help='Configuration file')
+    parser.add_argument('--spkr_conf', type=str, required=True,
+                        help='Speaker configuration file')
+    parser.add_argument('--featdir', type=str, required=True,
+                        help='Root directory of ground truth feature h5 files')
+    parser.add_argument('--outwavdir', type=str, required=True,
+                        help='Converted waveform directory')
+    parser.add_argument('--out', '-O', type=str,
+                        help='The output filename. '
+                             'If omitted, then output to sys.stdout')
+    parser.add_argument("--n_jobs", default=40, type=int,
+                        help="number of parallel jobs")
+    args = parser.parse_args()
+    
+    # logging info
+    logging.basicConfig(
+        level=logging.INFO,
+        stream=sys.stdout,
+        format="%(asctime)s (%(module)s:%(lineno)d) " "%(levelname)s: %(message)s",
+    )
+    
+    # load configure files
+    conf = load_yaml(args.conf)
+    spkr_conf = load_yaml(args.spkr_conf)
+    
+    # load converted files. If mcep, use h5; else, waveform
+    if conf["feat_type"] == "mcep":
+        converted_files = sorted(find_files(args.outwavdir, query="*.h5"))
+    else:
+        converted_files = sorted(find_files(args.outwavdir))
+
+    # load ground truth scp
+    scp = {}
+    featdir = Path(args.featdir) / conf["feature"]["label"]
+    gt_feats = open_featsscp(featdir / "eval" / "feats.scp")
+    
+    # Get and divide list
+    logging.info(f"number of utterances = {len(converted_files)}")
+    file_lists = np.array_split(converted_files[:100], args.n_jobs)
+    file_lists = [f_list.tolist() for f_list in file_lists]
+    
+    if args.out is None:
+        out = sys.stdout
+    else:
+        out = open(args.out, 'w', encoding='utf-8')
+
+    # multi processing
+    with mp.Manager() as manager:
```

Comment:
This is just a question. Do you have any special reason to use mp instead of joblib for multiprocessing?

Reply:
Yes. Because I want to keep a dictionary for saving the resulting scores asynchronously, but `joblib` does not have such a function.

Reply:
I see. I will check the `joblib` documentation.